### ISSUE TITLE
The rarity does not work for me

### ISSUE THREAD

**Issue by xddyyy** [HashLips/generative-art-node#25]
Is the rarity actually working? Im not good at coding especially Js but if I look through the code I cant see where the superrare is used?

**Comment by sdotwinter** [comment]
I am wondering the same. Is it just for metadata?

**Comment by petercottonmouth** [comment]
> I am wondering the same. Is it just for metadata?

Right, the metadata shows up with the correct titles but it doesn't seem to actually make anything more rare. I'm trying to figure out how to code rarity into the program and I'll let you know when I figure it out.

**Comment by vagueink** [comment]
@petercottonmouth it would be awesome if you can figure this out. Thanks!

**Comment by grantreighard** [comment]
The method I'm using is to add blank images to create the rarity within a directory. For instance, if I want a layer to be 1/100, I have 99 blanks along with the one image that contains something. This is just an artificial method for creating rarity (it works though), but the attribute will have a value (such as "blank35" when picking the 35th blank) in the metadata regardless of if the layer is blank or not.

**Comment by pguardiario** [comment]
> > I am wondering the same. Is it just for metadata?
> 
> Right, the metadata shows up with the correct titles but it doesn't seem to actually make anything more rare. I'm trying to figure out how to code rarity into the program and I'll let you know when I figure it out.

You just put _sr (super rare) or _r (rare) in the image filenames

**Comment by grantreighard** [comment]
> You just put _sr (super rare) or _r (rare) in the image filenames

This alone does not make anything more rare. If you have one layer directory with only one file, even if you label it _sr, every single output image will contain it.

**Comment by pguardiario** [comment]
> > You just put _sr (super rare) or _r (rare) in the image filenames
> 
> This alone does not make anything more rare. If you have a layer directory with only one file, even if you label it _sr, every single output image will contain it.

Yes, it only affects the metadata property. To change the math you need to do something in the `drawLayer` function
